_drafts/oryx/ssr.md

```diff
@@ -0,0 +1,132 @@
+## Introduction to SSR
+
+### Why SSR?
+
+Server-Side Rendering (SSR), including Static Site Generation (SSG) as a variant, has grown in popularity due to its ability to boost web application performance, facilitate effective Search Engine Optimization (SEO), social sharing, and improve Core Web Vitals (CWV). By delivering pre-rendered HTML from the server (or even CDN) to the client, SSR and SSG lead to quicker initial page load times, enhance the user experience, and can significantly improve CWV scores. SSG, in particular, pre-renders HTML at build time, resulting in static HTML, CSS, and JavaScript files that can be served directly from a CDN. It is a useful strategy for sites with content that does not change frequently, and can improve performance, scalability, and security.
+
+### SSR vs. CSR: Understanding the Differences
+
+**Speed & Performance**: With SSR, the browser receives pre-rendered HTML, reducing the time taken to display meaningful content.
+
+**SEO**: SSR is typically more favorable for SEO since search engine crawlers find it easier to index pre-rendered HTML content.
+
+**Social Media Integration**: SSR significantly improves integration with social providers like Facebook and Twitter, and bots like Slack. It enables the generation of link previews, rich snippets, and thumbnails, enhancing the visibility and appeal of shared content on these platforms.
+
+**User Experience**: By delivering pre-rendered content faster, SSR minimizes user waiting time, providing a superior user experience compared to CSR.
+
+**Resource Allocation**: While SSR enhances performance and user experience, it requires more server resources and processing power. CSR lightens server load by offloading rendering to the client, but at the cost of potentially increased load times and less effective SEO.
+
+
+### SSR: Pros and Cons
+
+| Benefits of SSR                                                   | Drawbacks of SSR    |
+|-------------------------------------------------------------------|---------------------|
+| Quicker initial page load times.                                  | Higher server resource usage. |
+| Enhanced SEO.                                                     | Increased development and deployment complexity. |
+| Improved user experience.                                         | Infrastructure concerns. |
+| Better social media integration.                                  | Potential for stale content. |
+
+#### Applicability of SSR
+
+While SSR offers numerous benefits, it may not be the best fit for every type of application, for example:
+
+- **B2B (Closed) Shops**: Public-facing SEO and quicker initial page load times offered by SSR may not significantly benefit these types of applications.
```

Comment:
What is a closed shop?

Reply:
replaced "(Closed)" with "restricted access"